matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixPatterns.kt

```diff
@@ -162,10 +162,14 @@ object MatrixPatterns {
         return order != null && order.length < 50 && order matches ORDER_STRING_REGEX
     }
 
-    fun candidateAliasFromRoomName(name: String): String {
-        return Regex("\\s").replace(name.lowercase(), "_").let {
-            "[^a-z0-9._%#@=+-]".toRegex().replace(it, "")
-        }
+    fun candidateAliasFromRoomName(roomName: String, domain: String): String {
+        return roomName.lowercase()
+                // Replace spaces by '_'
+                .let { Regex("\\s").replace(it, "_") }
```

Comment:
nit: these comments could be replaced by extensions with the same name as the comment (and potentially be reused)

```kotlin
roomName
  .lowercase()
  .replaceSpacesWithUnderscore()
  .removeInvalidRoomNameChars()
  .limitLength(length = MatrixConstants.maxAliasLocalPartLength(domain))


fun String.replaceSpacesWithUnderscore() = Regex("\\s").replace(this, "_")
fun String.removeInvalidRoomNameChars() = "[^a-z0-9._%#@=+-]".toRegex().replace(this, "")
fun String.limitLength(length: Int) = this.substring(0, length)
```

I'm more in the _self documenting code_ camp than adding comments to describe what the code does, where possible!

Reply:
👍 done in f385e7466238c101ae8afbd0ea519eb0f3291652